### DISCUSSION TITLE
General

### DISCUSSION THREAD

**Discussion by omodia** [mienaiyami/yomikiru#85] [General]
Found another cause of why it is [flagged as malware](https://github.com/mienaiyami/yomikiru/issues/80#issuecomment-1488839377).

### Steps to reproduce
- Unzip any yomikiru version archive.
- Unzipping dialogue appears in reader and then it flagged as malware.

Wanted to let you know about them.

**Comment by mienaiyami** [comment]
Sorry but there's nothing I can do about it since app is not digitally signed. 

> Unzipping dialogue appears in reader and then it flagged as malware.

Its because an unsigned app is trying to extract files in you system, so your antivirus will flag it of course

**Reply by omodia** [reply]
A solution, maybe,
- By making the app's unzipper to skip extracting '.exe, .scr, non - image files' wouldn't that solve the issue?
- Only make the app extract the image files supported by it. 
- For reference, in winrar settings one can specify the file extensions that can be excluded to extract for all archives being extracted by winrar.

**Reply by mienaiyami** [reply]
> Only make the app extract the image files supported by it

What do you mean by than?

And files cant be skipped, it may not look like it but .exe file is also replaced every version up.

**Reply by omodia** [reply]
> What do you mean by than?

- No, I am not talking about the app update / version update.
- I meant, is it not possible to add a rule to only extract (".jpg", ".jpeg", ".png", ".webp", ".svg", ".apng", ".gif", ".avif") from a zip, cbz archive.
- It is not flagged as antivirus, while trying to extract only image files, the problem occurs when the extracts an archive that has .exe file.
- So wouldn't this rule solve the issue.

> And files cant be skipped, it may not look like it but .exe file is also replaced every version up.

- Not that. But, when the application is uzipping a zip, cbz file to open manga. In that scenario. 
- A similar rule to skip non - image files just like you can see below in winrar.

https://user-images.githubusercontent.com/116664905/235167598-466697c1-0b6f-4082-8f60-fe8099053ffe.mp4

**Reply by mienaiyami** [reply]
My current unzipper does not have this option. I will look into it.

**Comment by omodia** [comment]
The second reason for virus / malware flagging is 'continous reloading' as I mentioned previously in an issue. If you remember.

- For this, limiting the number of times the app can be reloaded would solve the issue.
- So that, even if the key was accidentally stuck while using the keyboard, the app will not be reloaded continously.
- And will not be flagged as malware / virus.
- Example, Limiting the reload (CTRL+R) to - 1 time per 10 seconds.

These two are the only reasons for the app to be flagged as malware / virus.
